server/protocols/jmap/src/main/java/org/apache/james/jmap/model/MessageFactory.java

```diff
@@ -97,13 +109,21 @@ public Message fromMetaDataWithContent(MetaDataWithContent message) throws Mailb
                 .replyTo(fromAddressList(mimeMessage.getReplyTo()))
                 .size(message.getSize())
                 .date(message.getInternalDateAsZonedDateTime())
-                .textBody(messageContent.getTextBody().orElse(null))
-                .htmlBody(messageContent.getHtmlBody().orElse(null))
-                .preview(getPreview(messageContent))
+                .textBody(textBody)
```

Comment:
Ah and if not here we need these test cases : 
 - what happens to htmlBody when not here ? -> it should be null / absent in the final JSON
 - What heppens when testBody and htmlBody are both not here ? -> it should be both null / absent in the final JSON